### PR TITLE
Fixes to be more lenient on configs without users.

### PR DESCRIPTION
**Pull request by brendandburns**
This makes the library work with kubernetes-dind clusters.

### CODE REVIEW

File: src/config_types.ts [hunk 1]
```diff
@@ -101,7 +101,7 @@ function contextIterator(): u.ListIterator<any, Context> {
         if (!elt.context.cluster) {
             throw new Error(`contexts[${i}].context.cluster is missing`);
         }
-        if (!elt.context.user) {
+        if (elt.context.user === undefined) {
```
Comment (itowlson) [inline src/config_types.ts]: Not quite sure of the import of this change.  Is the intention to permit the empty string (or `null`), but still forbid `undefined`?

Comment (brendandburns) [reply src/config_types.ts]: That's what I was going for. TBH, I don't know what the "right" expectation is here.... I can test with a config with no user at all and see what `kubectl` does...

Comment (brendandburns) [reply src/config_types.ts]: Looks like kubectl is ok with it 100% missing, so I just removed this.

**Pull request merged**